motor/discovery/azure/subscriptions.go

```diff
@@ -45,8 +49,36 @@ func (client *Subscriptions) GetSubscriptions() ([]subscriptions.Subscription, e
 			return nil, err
 		}
 		for _, s := range page.Value {
-			subs = append(subs, *s)
+			if !skipSub(s, filter) {
+				subs = append(subs, *s)
+			}
 		}
 	}
 	return subs, nil
 }
+
+func skipSub(sub *subscriptions.Subscription, filter subscriptionsFilter) bool {
+	// anything explicitly specified in the list of includes means accept only from that list
+	if len(filter.include) > 0 {
+		for _, s := range filter.include {
+			if s == *sub.SubscriptionID {
+				return false
+			}
+		}
+		// didn't find it, so it must be skipped
+		return true
+	}
+
+	// if nothing explicitly meant to be included, then check whether
+	// it should be excluded
+	if len(filter.exclude) > 0 {
```

Comment:
If I read this code correctly it means that if you provide both include and exclude it won't work. I think that's good because it makes the code simpler. I do think that you should add a check in the CLI to make sure that we return an error if someone provides both flags at the same time